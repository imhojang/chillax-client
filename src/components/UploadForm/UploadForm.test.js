import React from 'react';
import { mount } from 'enzyme';
import UploadForm from './UploadForm';

describe('<UploadForm />', () => {
  // snapshot testing
  it('matches snapshot', () => {
    const wrapper = mount(<UploadForm />);
    expect(wrapper).toMatchSnapshot();
  });

  // props 접근
  it('props are passed down correctly', () => {
    const user = {
      name: 'John Smith',
      email: 'johnsmith@gmail.com',
      _id: '3djf481kashd381ksfjdn1',
    };
    const handleClose = jest.fn();
    const showUploadModal = jest.fn();
    const uploadSound = jest.fn();
    const wrapper = mount(
      <UploadForm
        handleClose={handleClose}
        showUploadModal={showUploadModal}
        uploadSound={uploadSound}
        user={user}
      />
    );
    expect(wrapper.props().user).toBe(user);
    expect(wrapper.props().handleClose).toBe(handleClose);
    expect(wrapper.props().showUploadModal).toBe(showUploadModal);
    expect(wrapper.props().uploadSound).toBe(uploadSound);
  });

  // DOM 확인
  it('renders texts on DOM correctly', () => {
    const wrapper = mount(<UploadForm />);

    const whiteNoiseSpanElem = wrapper.find('.white-noise-text');
    expect(whiteNoiseSpanElem.contains('white noise')).toBe(true);
  });

  it('click on cancel button closes upload modal', () => {
    const handleClose = jest.fn();
    const wrapper = mount(<UploadForm handleClose={handleClose} />);
    let cancelButton = wrapper.findWhere(
      node => node.type() === 'button' && node.text() === 'Cancel'
    );
    cancelButton.simulate('click');
  });
});
