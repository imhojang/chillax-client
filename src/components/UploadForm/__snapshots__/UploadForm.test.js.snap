// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadForm /> matches snapshot 1`] = `
<UploadForm>
  <div
    className="form-container"
  >
    <form
      className="upload-form"
      encType="multipart/form-data"
      onSubmit={[Function]}
    >
      <div
        className="sound-thumbnail-input-container"
      >
        <span>
          Select the type of sound you would like to upload:
        </span>
        <br />
        <label
          className="type-label"
        >
          <input
            name="type"
            onChange={[Function]}
            type="radio"
            value="white-noise"
          />
          <span
            className="white-noise-text"
          >
            white noise
          </span>
        </label>
        <label
          className="invisible"
          htmlFor="upload-sound"
        >
          <div>
            <p
              className="visible"
            >
              <br />
              You can upload a single sound for white noise.
              <br />
            </p>
            <p
              className="invisible"
            >
              <br />
              You can upload upto four sound for soundscape.
              <br />
            </p>
            <input
              accept="audio/*"
              id="upload-sound"
              multiple="multiple"
              name="sound"
              onChange={[Function]}
              type="file"
            />
          </div>
        </label>
        <label
          className="invisible"
          htmlFor="upload-thumbnail"
        >
          <div
            className="thumb"
          >
            <img
              alt=""
              id="thumbnail-preview"
              src="//:0"
            />
            <button
              className="visible"
              onClick={[Function]}
              type="button"
            >
              Upload image
            </button>
          </div>
          <p>
            <button
              className="invisible"
              id="replace-image-button"
              onClick={[Function]}
              type="button"
            >
              replace image
            </button>
            <input
              accept="image/*"
              id="upload-thumbnail"
              name="image"
              onChange={[Function]}
              type="file"
            />
          </p>
        </label>
      </div>
      <div
        className="title-description-container"
      >
        <label
          className="invisible"
          htmlFor="title-input"
        >
          <span
            className="required-field"
          >
            *
          </span>
          Title
          <br />
          <input
            id="title-input"
            name="title"
            onChange={[Function]}
            placeholder="enter title"
            required={true}
            type="text"
            value=""
          />
        </label>
        <label
          className="invisible"
          htmlFor="description-input"
        >
          <span
            className="required-field"
          >
            *
          </span>
          Description
          <br />
          <textarea
            cols="40"
            id="description-input"
            name="description"
            onChange={[Function]}
            placeholder="describe your sound here"
            required={true}
            rows="5"
            value=""
          />
        </label>
      </div>
      <div
        className="decision-buttons-container"
      >
        <div
          className="invisible"
        >
          <span
            className="required-field"
          >
            *
          </span>
          Required Fields
          <div
            className="save-cancel-button-container"
          >
            <button
              id="cancel-button"
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
            <input
              id="save-button"
              type="submit"
              value="Save"
            />
          </div>
        </div>
      </div>
    </form>
  </div>
</UploadForm>
`;
